### DISCUSSION TITLE
[Rust] Configuring NVChad for rust

### DISCUSSION THREAD

**Discussion by sananand007** [NvChad/NvChad#2594] [Q&A]
Hi Community, 

I am using NVChad for rust, I would like to know about some current custom configurations for autocomplete and typing help for using RUST on NVChad. 

Also if there are some more information on how to do `cargo` operations as well without starting a `terminal` that would be nice to know about. 

For example : 

I am trying to use the `Box<T>` type, but I do not see it recognized. Some stuff are, but a lot of generic stuff in rust are not coming up. I would like to have typing and auto completions as descriptive in `rust` as possible. 

<img width="912" alt="image" src="https://github.com/NvChad/NvChad/assets/13934022/655cd842-3f5d-4c4d-aec8-0ff8be973737">

**Comment by AJMacDuk** [comment]
Dreams of Code has a good [NvChad rust setup youtube video](https://youtu.be/mh_EJhH49Ms?si=9Jd9XV1oEeYvBIF9). There's a link to the configs in the video description.

I replaced rust tools with [rustaceanvim](https://github.com/mrcjkb/rustaceanvim) since rust-tools has been archived. Used the following plugin config for that:

```lua
{
    "mrcjkb/rustaceanvim",
    version = "^3",
    ft = { "rust" },
    config = function(_,_)
      vim.g.rustaceanvim = {
        server = {
          on_attach = function(client, buffer)
            require("core.utils").load_mappings("lspconfig", { buffer = buffer })
            require("nvchad.signature").setup(client)
          end,
        },
      }
    end
},
```
rustaceanvim can handle building via code actions.

Remember to also install rust-analyzer (you can do so via mason).

**Reply by siduck** [reply]
@AJMacDuk  just import nvchad's on_attach func and use  it as

server = { on_attach = on_attach } :sunglasses:

**Reply by AJMacDuk** [reply]
Thanks @siduck, worked like a charm.

**Reply by lekhanhtoan37** [reply]
In nvchad 2.5, they drop `load_mappings`
```
require("core.utils").load_mappings
```
So not worked in nvchad 2.5

**Reply by siduck** [reply]
@lekhanhtoan37  as its not needed,   just load mappings directly i.e vim.keymap.set? https://nvchad.com/docs/config/mappings

as for on_attach stuff, use the one which u import from nvchad.configs.lspconfig module
